src/slug.rs

```diff
@@ -0,0 +1,77 @@
+//! A trait (and implementation) to specify the anchor calculation.
+use std::{borrow::Cow, collections::HashMap};
+
+use once_cell::sync::Lazy;
+use regex::Regex;
+
+pub trait Slugify {
+    fn slugify(&mut self, str: &str) -> Cow<str>;
```

Comment:
The `Cow` here is currently not used but I think it makes sense to allow implementations to be more efficient in case a heading text doesn't need to be sluggified.

Reply:
I think a test that demonstrates this behaviour would be useful.

Reply:
Now that the lifetimes are explicit a test for the ability to return a borrowed Cow seems a bit redundant to me. Or what behavior were you talking about?